### DISCUSSION TITLE
403 error trying to access api routes

### DISCUSSION THREAD

**Discussion by esdee** [flareact/flareact#152] [Q&A] [answered]
I have a very simple test app with 1 api route and an index.tsx file

The api route `/pages/api/messages.tsx` consists of the following 
```
export default async (event) => {
  return {hello: 'world !!'};
};
```

In the `index.tsx` file I have the following code 
```
export async function getEdgeProps() {
  const endpoint = 'http://localhost:8080/api/messages';
  const response = await fetch(endpoint, {headers: {'Content-Type': 'application/json'}});
  console.log(JSON.stringify(response, null, 2));
  return {
    props: {
      message: `Status : ${response.status}`
    } as IndexProps,
  };
}
```

Attempting to `fetch` a response from /api/messages results in a `403` status. I had 2 questions:
1. How do I prevent the 403?
2. What is the correct way to refer to internal endpoints in /api so that this works locally and when deployed?
Thx!

**Comment by traffisco** [comment, marked as answer]
Omit the local host. Leave it on /api/.
If it doesn't work, post a link to reproduction repo please.

**Reply by esdee** [reply]
Thanks for the response. Changing the endpoint to `/api/messages` did not work. 
This generated a workers error.
The repo is here https://github.com/esdee/workers-test if you had any suggestions.

**Reply by tj-kev** [reply]
I would imagine this is because the "getEdgeProps" is running in the worker. So doesn't have any concept of localhost or a root path like "/api".

There also seems to be a restriction on workers calling workers through their full domain. Each layer called must be on a different domain. 

There is an answer about it on the cloudflare forums here: https://community.cloudflare.com/t/is-a-worker-allowed-to-make-requests-to-another-worker/194733

I would recommend moving the logic contained within the api endpoint to the getEdgeProps method and save yourself the extra api call.

**Reply by traffisco** [reply]
Oh, good catch. Didn't noticed that's called from within getEdgeProps. No point doing that, just import the function and call it.
Use /api endpoint if from client(useEffect)

**Reply by tj-kev** [reply]
Weirdly it does work from the workers quick edit screen in the cloudflare console. Strange. Just one of those quirks I guess.

**Reply by esdee** [reply]
Thank you for the replies, this worked.
